resolver/request.py

```diff
@@ -39,15 +61,23 @@ def resolve(self):
         :rtype: str
         """
 
-        url = (
-            self.argument.get("url")
-            if isinstance(self.argument, dict)
-            else self.argument
-        )
+        args = self.argument
+        url = args
 
-        if checkers.is_url(url):
-            response = self._make_request(url)
-        else:
+        if isinstance(args, dict):
+            validate(instance=args, schema=RESOLVER_ARGS_SCHEMA)
+            url = args.get("url")
+
+        if not checkers.is_url(url):
             raise InvalidResolverArgumentValueError(f"Invalid argument: {url}")
 
+        auth = None
+        if isinstance(args, dict) and "auth" in args:
+            auth_type = args.get("auth").lower()
+            if auth_type == "basic":
+                user = args.get("user")
+                password = args.get("password")
```

Comment:
Why are you using `.get()` here? Are these keys optional if `auth` is set to "basic"? I think there should be some sort of error if either of these are not set.

Reply:
yes, you are correct.  values for these keys need to be set if auth is basic.  the schema validation checks for this and there are tests for.

Reply:
In that case, it would be clearer to readers of this code if you used `args["password"]` and so on instead of `.get()`, which is only used when a key is optional.

Reply:
done.